### PR TITLE
Add testProvidersMocked for unit testing

### PR DESCRIPTION
**Pull request by meyskens**
Signed-off-by: Maartje Eyskens <maartje@eyskens.me>

### CODE REVIEW

**Comment by meyskens** [comment]
Of course tests fail... will push a fix

**Comment by rporres** [comment]
So you always create the mocked provider but you need to call specifically it from the tests. Looks good. But I'm not sure if normal acceptance tests will complain as `testProviders["online"] = Provider()` hasn't been created. Or are they completely ignored?

File: provider/provider_test.go [hunk 1]
```diff
@@ -10,18 +10,18 @@ import (
 )
 
 var testProviders = map[string]terraform.ResourceProvider{}
+var testProvidersMocked = map[string]terraform.ResourceProvider{}
```
Comment (rporres) [inline provider/provider_test.go]: I think the standard way of calling this is `testAccProviders` and `testMockProviders`

File: provider/provider_test.go [hunk 2]
```diff
@@ -10,18 +10,18 @@ import (
 )
 
 var testProviders = map[string]terraform.ResourceProvider{}
```
Comment (rporres) [inline provider/provider_test.go]: The standard naming for this is testAccProviders (at least this is what I'm seeing in other providers and doc). Could you rename it too?

**Pull request merged**